PR for SRVLOGIC-530: Add the section to upgrade Serverless Logic Operator from 1.35.0 to 1.36.0

kaldesai opened the pull request:
**Affected versions:** 
- `serverless-docs-1.36`
- `serverless-docs-1.35`

**Tracking JIRA:** https://issues.redhat.com/browse/SRVLOGIC-530

**Doc preview:** [Upgrading Serverless Logic Operator from the version 1.35.0 to 1.36.0](https://93332--ocpdocs-pr.netlify.app/openshift-serverless/latest/serverless-logic/serverless-logic-managing-upgrades/serverless-logic-upgrading-operator-from-1.35-to-1.36)

**Reviews:**
- [x] QE has approved this change.
<!--- QE approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->
- [x] SME has approved this change.
<!--- SME approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

Commit:
SRVLOGIC-530-upgrade: Added a section about upgrading from 1.35 to 1.36 in the OSL docs

kaldesai commented:
@ricardozanini, @gmunozfe, @wmedvede, The PR is now ready for review. Can you please take a look and verify the content? Thank you!

kaldesai commented:
@briandooley FYI

wmedvede commented on modules/serverless-logic-upgrade-1.36-finalizing-job-service.adoc:
```diff
@@ -0,0 +1,39 @@
+[id="serverless-logic-upgrade-1.36-finalizing-job-service_{context}"]
+= Finalizing the Job Service upgrade
+
+After the {ServerlessOperatorName} is upgraded to version 1.36.0, a new ReplicaSet for the Job Service component is created automatically. To clean up older resources, you can delete the Job Service ReplicaSets created by the previous 1.35.0 version.
```
```suggestion
After the {ServerlessOperatorName} is upgraded to version 1.36.0 you can optionally delete the old ReplicaSet`.
```

wmedvede commented on modules/serverless-logic-upgrade-1.36-finalizing-job-service.adoc:
```diff
@@ -0,0 +1,39 @@
+
+.Procedure
+
+. List all ReplicaSets for the Job Service by running the following command:
```
```suggestion
. (Optional) Identify and the delete the old `ReplicaSet` running the following commands:
```

kaldesai replied:
@wmedvede 

For listing, the command is different. 
`oc get replicasets -o custom-columns=Name:metadata.name,Image:spec.template.spec.containers[*].image -n <target_namespace>`

For deleting the command is different: 
`oc delete replicaset <old_replicaset_name> -n <target_namespace>`

Then why are we removing these steps?

wmedvede replied:
@kaldesai the case of the jobs service is analogous to the DataIndex ReplicasSet removal.
See attached screenshot.

People can optionally delete.

wmedvede commented on modules/serverless-logic-upgrade-1.36-finalizing-job-service.adoc:
```diff
@@ -0,0 +1,39 @@
+----
++
+.Example output
+[source,sql]
```
```suggestion
[source,sql,subs="verbatim,quotes"]
```

wmedvede commented on modules/serverless-logic-upgrade-1.36-finalizing-job-service.adoc:
```diff
@@ -0,0 +1,39 @@
+[source,sql]
+----
+Name                                                Image
+sonataflow-platform-jobs-service-1111111111         registry.redhat.io/openshift-serverless-1/logic-jobs-service-postgresql-rhel8:1.35.0
```
```suggestion
*sonataflow-platform-jobs-service-1111111111         registry.redhat.io/openshift-serverless-1/logic-jobs-service-postgresql-rhel8:1.35.0*
```

wmedvede replied:
I have added the *blabla* with hope that this particular line is displayed in bold in the produced outptut.
Help users to identify which is the like that must be used as reference for deletion.

wmedvede commented on modules/serverless-logic-upgrade-1.36-finalizing-job-service.adoc:
```diff
@@ -0,0 +1,39 @@
+sonataflow-platform-jobs-service-2222222222         registry.redhat.io/openshift-serverless-1/logic-jobs-service-postgresql-rhel8:1.36.0
+----
+
+. Identify and delete the old ReplicaSet associated with version 1.35.0 by running the following command:
```
```suggestion
```

wmedvede commented on modules/serverless-logic-upgrade-1.36-redeploying-workflows-with-dev-profile.adoc:
```diff
@@ -0,0 +1,20 @@
+
+.Procedure
+
+. Ensure all required resources are restored, including `SonataFlow` CRDs, `ConfigMaps` with `application.properties`, or any other related custom configurations.
```
```suggestion
. Ensure all required Kubernetes resources, including the `ConfigMap` with the `application.properties`, are restored before redeploying the workflow.
```

wmedvede commented on modules/serverless-logic-upgrade-1.36-restoring-workflows-with-preview-profile.adoc:
```diff
@@ -0,0 +1,20 @@
+
+.Procedure
+
+. Ensure all required resources are recreated, including `SonataFlow` CRDs, `ConfigMaps` with `application.properties`, or any other related custom configurations before you redeploy the workflow.
```
```suggestion
. Ensure all required Kubernetes resources, including the `ConfigMap` with the `application.properties`, are restored before redeploying the workflow.
```

wmedvede commented on modules/serverless-logic-upgrade-1.36-restoring-workflows-with-preview-profile.adoc:
```diff
@@ -0,0 +1,20 @@
+[id="serverless-logic-upgrade-1.36-restoring-workflows-with-preview-profile_{context}"]
+= Restoring workflows with the preview profile
+
+Workflows with the `preview` profile require an additional configuration step before being redeployed.
```
```suggestion
After the upgrade, you must redeploy workflows that use the `preview` profile and any associated Kubernetes resources.
```

wmedvede commented on modules/serverless-logic-upgrade-1.36-scaling-up-workflows-with-gitops-profile.adoc:
```diff
@@ -0,0 +1,34 @@
+
+.Procedure
+
+. Modify the `my-workflow.yaml` CRD and scale up each workflow to `one` before upgrading as shown in the following example:
```
```suggestion
. Modify the `my-workflow.yaml` CR and scale up each workflow to `one` as shown in the following example:
```

wmedvede commented on modules/serverless-logic-upgrade-1.36-scaling-up-workflows-with-gitops-profile.adoc:
```diff
@@ -0,0 +1,34 @@
+    replicas: 1
+----
+
+. Apply the updated CRD by running the following command:
```
```suggestion
. Apply the updated CR by running the following command:
```

wmedvede commented on modules/serverless-logic-upgrade-1.36-scaling-up-workflows-with-gitops-profile.adoc:
```diff
@@ -0,0 +1,34 @@
+$ oc apply -f <my-workflow.yaml> -n <target_namespace>
+----
+
+. (Optional) Scale the workflow back to `1` by running the following command:
```
```suggestion
. (Alternative) Scale the workflow back to `1` by running the following command:
```

kaldesai replied:
@wmedvede 
According to the IBM & OpenShift standard, we have to use (Optional) here.

kaldesai commented on modules/serverless-logic-upgrade-1.36-scaling-down-workflows-with-gitops-profile.adoc:
```diff
@@ -0,0 +1,34 @@
+$ oc apply -f <my-workflow.yaml> -n <target_namespace>
+----
+
+. (Optional) Scale the workflow to `0` by running the following command:
```
According to the IBM & OpenShift standard guide, we have to use (Optional) here.

kaldesai commented on modules/serverless-logic-upgrade-1.36-finalizing-data-index.adoc:
```diff
@@ -0,0 +1,72 @@
+sonataflow-platform-data-index-service-2222222222   registry.redhat.io/openshift-serverless-1/logic-data-index-postgresql-rhel8:1.36.0
+----
+
+. Identify the old Data Index ReplicaSet (with version 1.35.0) and delete it by running the following command:
```
@wmedvede 

dont we need this command? 

```
Identify the old Data Index ReplicaSet (with version 1.35.0) and delete it by running the following command:
+
[source,terminal]
----
$ oc delete replicaset <old_replicaset_name> -n <target_namespace>
----
```

wmedvede replied:
The removal of the Old ReplicaSet is optional.

kaldesai commented:
@wmedvede I have incorporated you suggestions with the doubts & answers. Please take a look

wmedvede commented on modules/serverless-logic-upgrade-1.36-finalizing-data-index.adoc:
```diff
@@ -0,0 +1,72 @@
++
+After deletion, a new trigger compatible with {ServerlessLogicProductName} 1.36.0 is automatically created.
+
+. Verify the new ReplicaSet exists by listing all ReplicaSets by running the following command:
```
@kaldesai  :point_up:  I think this is missing. Here we say that the deletion of the old ReplicaSet is optional and explain how to do it. Son in this optional step we explain two things, how to identify the old RS, and how to remove.

wmedvede replied:
The aim of this optional removal is to look like this, maybe we have a better way to organize, but this is the concept.

![image](https://github.com/user-attachments/assets/404de1e5-57f7-422d-b2aa-7c4d0dfd05dc)
```suggestion
. Verify the new ReplicaSet exists by listing all ReplicaSets by running the following command:
```

wmedvede commented on modules/serverless-logic-upgrade-1.36-finalizing-job-service.adoc:
```diff
@@ -0,0 +1,39 @@
+[id="serverless-logic-upgrade-1.36-finalizing-job-service_{context}"]
+= Finalizing the Job Service upgrade
+
+After the {ServerlessOperatorName} is upgraded to version 1.36.0 you can optionally delete the old ReplicaSet`.
```
```suggestion
After the {ServerlessOperatorName} is upgraded to version 1.36.0 you can optionally delete the old `ReplicaSet`.
```

wmedvede replied:
:point_up: nitpick

wmedvede reviewed (changes requested):
Hi @kaldesai , I have added a couple of nitpicks and some clarifiations.
Feel free to ping me if I couldn't clarify well.

kaldesai commented:
@wmedvede I have added your suggestions. Can you please take a look at the drafts once again? Thank you!

wmedvede commented on modules/serverless-logic-upgrade-1.36-finalizing-job-service.adoc:
```diff
@@ -0,0 +1,39 @@
+
+After the {ServerlessOperatorName} is upgraded to version 1.36.0 you can optionally delete the old `ReplicaSet`.
+
+.Procedure
```
@kaldesai 

For the jobs service we should apply the same pattern as for the Data Index
:point_down: 
![image](https://github.com/user-attachments/assets/6a5bdb69-0320-4b6b-ba3b-6993f0609ff8)

wmedvede reviewed (approved):
Great work, many thanks!

gmunozfe reviewed (commented):
Some minor comments, please check @kaldesai

wmedvede commented on modules/serverless-logic-upgrade-1.36-restoring-workflows-with-preview-profile.adoc:
```diff
@@ -0,0 +1,20 @@
+After the upgrade, you must redeploy workflows that use the `preview` profile and any associated Kubernetes resources.
+
+.Procedure
+
```
@kaldesai  @gmunozfe

I think guys we don't need this because we are treating the update from 1.35.0 to 1.36.0, meaning that, If you are using persistence in 1.35.0 you must already have been set that  kie.flyway.enabled=true in your workflows.

That  kie.flyway.enabled property, was introduced for updates from 1.34.0  to 1.35.0

wmedvede replied:
@domhanak  In the migration guide, I believe that no. I think it can confuse people that are just migrating their installation in 1.35.0 to 1.35.0

From the migration guide point of view, I believe it should be like this.

I have workflows already created in 1.35.0?

ok, if you used persistence, it means that you have already created and configured them properly. And, from the migration perspective you don¡'t need do add any configuration.

I want to start a WF from scratch?

where, in 1.35.0? I need to read the workflows configuration section, etc,  and know how to configure the persistence.

where, in 1.36.0? Same as :point_up: , but if I am already in 1.36.0, a migration o a brand new installation was already performed.

kaldesai commented on serverless-logic/serverless-logic-managing-upgrades/serverless-logic-upgrading-operator-from-1.35-to-1.36.adoc:
```diff
@@ -0,0 +1,72 @@
+
+include::modules/serverless-logic-upgrade-1.36-backing-up-job-service-database.adoc[leveloffset=+2]
+
+include::modules/serverless-logic-upgrading-1.36-osl-operator.adoc[leveloffset=+1]
```
That is correct. I am keeping the leveloffset for +1 because I want to keep the following sections at the same level. I hope that answers your question. :) 

- Preparing for the upgrade
- Upgrading the OpenShift Serverless Logic Operator
- Finalizing the upgrade

kaldesai commented:
@gmunozfe, I have added your suggestions and answered your question regarding the level offset. Can you please take another round of review?

kaldesai commented:
/label peer-review-needed

xenolinux commented on modules/serverless-logic-upgrade-1.36-deleting-workflows-with-dev-profile.adoc:
```diff
@@ -0,0 +1,20 @@
+
+:_mod-docs-content-type: PROCEDURE
+[id="serverless-logic-upgrade-1.36-deleting-workflows-with-dev-profile_{context}"]
+= Deleting workflows with dev profile
```
```suggestion
= Deleting workflows with the dev profile
```

xenolinux commented on modules/serverless-logic-upgrade-1.36-deleting-migrating-workflows-with-preview-profile.adoc:
```diff
@@ -0,0 +1,22 @@
+[id="serverless-logic-upgrade-1.36-deleting-migrating-workflows-with-preview-profile_{context}"]
+= Deleting workflows with the preview profile
+
+Before upgrading the Operator, you must delete workflows running with the `preview` profile. When the upgrade is completed, you must redeploy the workflows.
```
```suggestion
Before upgrading the Operator, you must delete workflows running with the `preview` profile. When the upgrade is complete, you must redeploy the workflows.
```

xenolinux commented on modules/serverless-logic-upgrade-1.36-finalizing-job-service.adoc:
```diff
@@ -0,0 +1,37 @@
+
+.Procedure
+
+* (Optional) Identify the old `ReplicaSet` and delete it by running the following commands:
```
```suggestion
* (Optional) Identify the old `ReplicaSet` resource and delete it by running the following commands:
```

Same comment about using only a single command per code block.

xenolinux commented on modules/serverless-logic-upgrade-1.36-redeploying-workflows-with-dev-profile.adoc:
```diff
@@ -0,0 +1,20 @@
+
+.Procedure
+
+. Ensure that all required Kubernetes resources, including the `ConfigMap` with the `application.properties`, are restored before redeploying the workflow.
```
```suggestion
. Ensure that all required Kubernetes resources, including `ConfigMap` with the `application.properties` field, are restored before redeploying the workflow.
```

xenolinux commented on modules/serverless-logic-upgrade-1.36-redeploying-workflows-with-dev-profile.adoc:
```diff
@@ -0,0 +1,20 @@
++
+[source,terminal]
+----
+$ oc apply -f <my-workflow.yaml> -n <target_namespace>
```
```suggestion
$ oc apply -f <workflow_name>.yaml -n <target_namespace>
```

xenolinux commented on modules/serverless-logic-upgrade-1.36-restoring-workflows-with-preview-profile.adoc:
```diff
@@ -0,0 +1,20 @@
+
+.Procedure
+
+. Ensure that all required Kubernetes resources, including the `ConfigMap` with the `application.properties`, are restored before redeploying the workflow.
+
+. Redeploy the workflow by running the following command: 
++
+[source,terminal]
+----
+$ oc apply -f <my-workflow.yaml> -n <target_namespace>
+----
```
Same suggestions as earlier

xenolinux commented on modules/serverless-logic-upgrade-1.36-scaling-down-workflows-with-gitops-profile.adoc:
```diff
@@ -0,0 +1,34 @@
+[id="serverless-logic-upgrade-1.36-scaling-down-workflows-with-gitops-profile_{context}"]
+= Scaling down workflows with the gitops profile
+
+Before upgrading the Operator, you must scale down workflows running with the `gitops` profile, and scale them up again after the upgrade is completed.
```
```suggestion
Before upgrading the Operator, you must scale down workflows running with the `gitops` profile, and scale them up again after the upgrade is complete.
```

xenolinux commented on modules/serverless-logic-upgrade-1.36-scaling-down-workflows-with-gitops-profile.adoc:
```diff
@@ -0,0 +1,34 @@
+
+.Procedure
+
+. Modify the `my-workflow.yaml` CR and scale down each workflow to `zero` before upgrading as shown in the following example:
```
expand CR

xenolinux commented on modules/serverless-logic-upgrade-1.36-scaling-down-workflows-with-gitops-profile.adoc:
```diff
@@ -0,0 +1,34 @@
+    replicas: 0
+----
+
+. Apply the updated `my-workflow.yaml' CR by running the following command:
```
```suggestion
. Apply the updated `my-workflow.yaml` CR by running the following command:
```

xenolinux commented on modules/serverless-logic-upgrade-1.36-scaling-down-workflows-with-gitops-profile.adoc:
```diff
@@ -0,0 +1,34 @@
++
+[source,terminal]
+----
+$ oc apply -f <my-workflow.yaml> -n <target_namespace>
```
```suggestion
$ oc apply -f my-workflow.yaml -n <target_namespace>
```

use `<  >` only for placeholders.

xenolinux commented on modules/serverless-logic-upgrade-1.36-scaling-down-workflows-with-gitops-profile.adoc:
```diff
@@ -0,0 +1,34 @@
++
+[source,terminal]
+----
+$ oc patch sonataflow <my-workflow> -n <target_namespace> --type=merge -p '{"spec": {"podTemplate": {"replicas": 0}}}'
```
```suggestion
$ oc patch sonataflow my-workflow -n <target_namespace> --type=merge -p '{"spec": {"podTemplate": {"replicas": 0}}}'
```

If  <my-workflow> is an example value then `< >` not required

xenolinux commented on modules/serverless-logic-upgrade-1.36-scaling-up-workflows-with-gitops-profile.adoc:
```diff
@@ -0,0 +1,34 @@
+[id="serverless-logic-upgrade-1.36-scaling-up-workflows-with-gitops-profile_{context}"]
+= Scaling up workflows with the gitops profile
+
+Workflows with the `gitops` profile that were previously scaled down must be scaled back up to continue operation.
```
```suggestion
To continue operation, you must scale up workflows that you previously scaled down with the `gitops` profile.
```

- Active voice

xenolinux commented on modules/serverless-logic-upgrade-1.36-scaling-up-workflows-with-gitops-profile.adoc:
```diff
@@ -0,0 +1,34 @@
+
+.Procedure
+
+. Modify the `my-workflow.yaml` CR and scale up each workflow to `one` as shown in the following example:
```
```suggestion
. Modify the `my-workflow.yaml` CR and scale up each workflow to `1` as shown in the following example:
```

- Expand CR for its first instance

xenolinux commented on modules/serverless-logic-upgrade-1.36-scaling-up-workflows-with-gitops-profile.adoc:
```diff
@@ -0,0 +1,34 @@
++
+[source,yaml]
+----
+spec:
+  podTemplate:
+    replicas: 1
```
https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#signs-and-symbols-in-code-blocks 

Consider applying this change to all of the other similar places in this PR

> For snippets or sections of a file, use an ellipsis (…​) to show that the file continues before or after the quoted block. For YAML, use ('#…​') instead.

xenolinux commented on modules/serverless-logic-upgrade-1.36-scaling-up-workflows-with-gitops-profile.adoc:
```diff
@@ -0,0 +1,34 @@
++
+[source,terminal]
+----
+$ oc apply -f <my-workflow.yaml> -n <target_namespace>
```
```suggestion
$ oc apply -f my-workflow.yaml -n <target_namespace>
```

Same comment as earlier

xenolinux commented on modules/serverless-logic-upgrade-1.36-scaling-up-workflows-with-gitops-profile.adoc:
```diff
@@ -0,0 +1,34 @@
++
+[source,terminal]
+----
+$ oc patch sonataflow <my-workflow> -n <target_namespace> --type=merge -p '{"spec": {"podTemplate": {"replicas": 1}}}'
```
```suggestion
$ oc patch sonataflow <workflow_name> -n <target_namespace> --type=merge -p '{"spec": {"podTemplate": {"replicas": 1}}}'
```

If <my-workflow> is the workflow name to be entered by users, consider the placeholder

xenolinux commented on modules/serverless-logic-upgrading-1.36-osl-operator.adoc:
```diff
@@ -0,0 +1,54 @@
+* You have {ServerlessLogicOperatorName} installed on your cluster.
+* You have access to a {ServerlessLogicProductName} project with the appropriate roles and permissions to create applications and other workloads in {ocp-product-title}.
+* You have access to the *OpenShift Management Console* for Operator upgrades.
+* You have installed the OpenShift CLI (`oc`).
```
- Consider using an attribute for OpenShift CLI (`oc`).

xenolinux commented on modules/serverless-logic-upgrading-1.36-osl-operator.adoc:
```diff
@@ -0,0 +1,54 @@
+.Procedure
+
+. In the web console, navigate to the *Operators* -> *OperatorHub* -> *Installed Operator* page.
+. Select the `openshift-serverless-logic` namespace from the *Installed Namespace*.
```
```suggestion
. Select the `openshift-serverless-logic` namespace from the *Installed Namespace*.
```

- the *Installed Namespace* drop down list? (Missing noun after an article)

xenolinux commented on modules/serverless-logic-upgrading-1.36-osl-operator.adoc:
```diff
@@ -0,0 +1,54 @@
+
+. In the web console, navigate to the *Operators* -> *OperatorHub* -> *Installed Operator* page.
+. Select the `openshift-serverless-logic` namespace from the *Installed Namespace*.
+. In the list of installed operators, find and click the {ServerlessLogicOperatorName}.
```
```suggestion
. In the list of installed Operators, find and click the {ServerlessLogicOperatorName} name.
```

Is "{ServerlessLogicOperatorName}" a name? (missing noun after an article)

xenolinux commented on modules/serverless-logic-upgrading-1.36-osl-operator.adoc:
```diff
@@ -0,0 +1,54 @@
+. Select the `openshift-serverless-logic` namespace from the *Installed Namespace*.
+. In the list of installed operators, find and click the {ServerlessLogicOperatorName}.
+. In the Operator details page, click on the *Subscription* tab. Click *Edit Subscription*.
+. In the *Upgrade status*, click the *Upgrade available* link.
```
```suggestion
. In the *Upgrade status* tab, click the *Upgrade available* link.
```

Is  *Upgrade status* a tab? (missing noun after an article)

xenolinux commented on modules/serverless-logic-upgrading-1.36-osl-operator.adoc:
```diff
@@ -0,0 +1,54 @@
+. In the list of installed operators, find and click the {ServerlessLogicOperatorName}.
+. In the Operator details page, click on the *Subscription* tab. Click *Edit Subscription*.
+. In the *Upgrade status*, click the *Upgrade available* link.
+. Click the *Preview install plan* and *Approve* to start the update.
```
Same comment - missing noun

xenolinux commented on modules/serverless-logic-upgrading-1.36-osl-operator.adoc:
```diff
@@ -0,0 +1,54 @@
+
+.Verification
+
+. To verify the new Operator version is installed, run the following command: 
```
```suggestion
* To verify that the new Operator version is installed, run the following command: 
```

`*` because it's a single step

xenolinux commented on modules/serverless-logic-verifying-the-1.36.0-upgrade.adoc:
```diff
@@ -0,0 +1,41 @@
+[id="serverless-logic-verifying-the-1.36.0-upgrade_{context}"]
+= Verifying the 1.36.0 upgrade
+
+After restoring workflows and services, it is essential to verify that the upgrade was successful and that all components are functioning as expected.
```
```suggestion
After restoring workflows and services, verify that the upgrade was successful and all components are functioning as expected.
```

or "services, **you must** verify that....."

gmunozfe commented:
> @gmunozfe, I have added your suggestions and answered your question regarding the level offset. Can you please take another round of review?

Now looks good to me, great work @kaldesai !

xenolinux commented on serverless-logic/serverless-logic-managing-upgrades/serverless-logic-upgrading-operator-from-1.35-to-1.36.adoc:
```diff
@@ -0,0 +1,72 @@
+
+toc::[]
+
+This section provides step-by-step instructions to upgrade the {ServerlessLogicOperatorName} from version 1.35.0 to 1.36.0. The upgrade process involves preparing the existing workflows and services, updating the Operator, and restoring the workflows after the upgrade.
```
```suggestion
You can upgrade the {ServerlessLogicOperatorName} from version 1.35.0 to 1.36.0. The upgrade process involves preparing the existing workflows and services, updating the Operator, and restoring the workflows after the upgrade.
```

> Avoid self-referential language, such as "This topic covers…​" or "Use this procedure to…​".
- https://redhat-documentation.github.io/supplementary-style-guide/#shortdesc

xenolinux commented on serverless-logic/serverless-logic-managing-upgrades/serverless-logic-upgrading-operator-from-1.35-to-1.36.adoc:
```diff
@@ -0,0 +1,72 @@
+
+[NOTE]
+====
+Different workflow profiles require different upgrade steps. Carefully follow the instructions for each profile.
```
```suggestion
Different workflow profiles require different upgrade steps. Follow the instructions for each profile carefully.
```

xenolinux reviewed (commented):
Left a few suggestions; otherwise looks good

kaldesai commented on modules/serverless-logic-upgrade-1.36-backing-up-job-service-database.adoc:
```diff
@@ -0,0 +1,15 @@
+// * serverless-logic/serverless-logic-upgrading-operator-from-1.35-to-1.36
+
+
+:_mod-docs-content-type: PROCEDURE
```
I initially structured “Backing up the Jobs Service database” as a procedure module because it involves a specific required action, taking a full backup before the upgrade, and outlines concrete steps, even if minimal. :|

ricardozanini commented on modules/serverless-logic-upgrade-1-36-deleting-migrating-workflows-with-preview-profile.adoc:
```diff
@@ -0,0 +1,22 @@
++
+[source,terminal]
+----
+$ oc delete -f <workflow_name>.yaml -n <target_namespace>
```
```suggestion
$ oc delete -f <workflow_name> -n <target_namespace>
```

Don't need to add the file extension.

wmedvede replied:
In this case, I think we must remove the -f  and type  `oc delete workflow <workflow_name> -n <target_namespace> `

@kaldesai   @ricardozanini

ricardozanini commented on modules/serverless-logic-upgrade-1-36-deleting-workflows-with-dev-profile.adoc:
```diff
@@ -0,0 +1,20 @@
++
+[source,terminal]
+----
+$ oc delete -f <workflow_name>.yaml -n <target_namespace>
```
```suggestion
$ oc delete -f <workflow_name> -n <target_namespace>
```

Same thing.

kaldesai commented:
/label merge-review-needed

bergerhoffer commented:
/label merge-review-in-progress

bergerhoffer commented on modules/serverless-logic-upgrade-1-36-finalizing-data-index.adoc:
```diff
@@ -0,0 +1,72 @@
++
+After deletion, a new trigger compatible with {ServerlessLogicProductName} 1.36.0 is automatically created.
+
+. (Optional) Identify the old `ReplicaSet` resource by running the following command:
```
For all of these optional steps, per the ISG, they should be written as "Optional: "

```suggestion
. Optional: Identify the old `ReplicaSet` resource by running the following command:
```

Screenshot from ISG:

![image](https://github.com/user-attachments/assets/937c1075-7648-4748-b4c0-0fd544a0923f)

bergerhoffer commented on modules/serverless-logic-upgrade-1-36-finalizing-job-service.adoc:
```diff
@@ -0,0 +1,39 @@
+----
++
+.Example output
+[source,sql,subs="verbatim,quotes"]
```
This should be source,terminal, not source,sql. The contents of the code block is output from the terminal, not SQL.

```suggestion
[source,terminal,subs="verbatim,quotes"]
```

bergerhoffer commented on modules/serverless-logic-upgrading-1-36-osl-operator.adoc:
```diff
@@ -0,0 +1,54 @@
+----
+.Expected output
++
+[source,sql]
```
```suggestion
[source,terminal]
```

bergerhoffer commented on modules/serverless-logic-verifying-the-1-36-0-upgrade.adoc:
```diff
@@ -0,0 +1,41 @@
+----
+.Expected output
++
+[source,sql]
```
```suggestion
[source,terminal]
```

bergerhoffer commented on modules/serverless-logic-upgrade-1-36-finalizing-data-index.adoc:
```diff
@@ -0,0 +1,72 @@
+----
++
+.Example output
+[source,sql,subs="verbatim,quotes"]
```
```suggestion
[source,terminal,subs="verbatim,quotes"]
```

bergerhoffer commented on modules/serverless-logic-upgrade-1-36-finalizing-data-index.adoc:
```diff
@@ -0,0 +1,72 @@
+----
++
+.Example output
+[source,sql,subs="verbatim,quotes"]
```
```suggestion
[source,terminal,subs="verbatim,quotes"]
```

bergerhoffer commented on modules/serverless-logic-upgrade-1-36-finalizing-data-index.adoc:
```diff
@@ -0,0 +1,72 @@
+[source,sql,subs="verbatim,quotes"]
+----
+Name                                                Image
+*sonataflow-platform-data-index-service-1111111111   registry.redhat.io/openshift-serverless-1/logic-data-index-postgresql-rhel8:1.35.0*
```
I'd remove this bolding, we don't do that on OCP docs as far as I know, and I don't think it's even working. Same for the other place you do it in another file.

bergerhoffer commented on modules/serverless-logic-upgrade-1-36-deleting-migrating-workflows-with-preview-profile.adoc:
```diff
@@ -0,0 +1,22 @@
+
+. If you are using persistence, back up the workflow database and ensure the backup includes both database objects and table data.
+
+. Ensure you have a backup of all necessary Kubernetes resources, including `SonataFlow` custom resources (CRs), `ConfigMaps`, or any other related custom configurations.
```
When mentioning the literal object, `ConfigMap` can't be plural. So you can either say `ConfigMap` resources, or you could generically say "config maps". See our guidelines on writing about API objects here: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#api-object-formatting

```suggestion
. Ensure you have a backup of all necessary Kubernetes resources, including `SonataFlow` custom resources (CRs), `ConfigMap` resources, or any other related custom configurations.
```

kaldesai replied:
Noted!

bergerhoffer commented on modules/serverless-logic-upgrade-1-36-finalizing-data-index.adoc:
```diff
@@ -0,0 +1,72 @@
+$ oc delete replicaset <old_replicaset_name> -n <target_namespace>
+----
++
+.Example command as per the generated example output
```
Per the ISG, shouldn't use "per"

You could either say "Example command based no the example output", or you could drop the heading and use a lead-in sentence to it

(Same with another module where you use this same wording)

bergerhoffer commented on modules/serverless-logic-upgrade-1-36-finalizing-job-service.adoc:
```diff
@@ -0,0 +1,39 @@
+sonataflow-platform-jobs-service-2222222222         registry.redhat.io/openshift-serverless-1/logic-jobs-service-postgresql-rhel8:1.36.0
+----
++
+. (Optional) Delete your old `ReplicaSet` resource by running the following command:
```
The whole procedure is optional, but this step isn't optional if you want to do the procedure. I don't think that either of these steps should be marked optional

kaldesai replied:
This makes sense

bergerhoffer commented on modules/serverless-logic-upgrade-1-36-scaling-down-workflows-with-gitops-profile.adoc:
```diff
@@ -0,0 +1,35 @@
+
+.Procedure
+
+. Modify the `my-workflow.yaml` custom resources (CR) and scale down each workflow to `zero` before upgrading as shown in the following example:
```
Should be the number if you're putting it into literals

```suggestion
. Modify the `my-workflow.yaml` custom resources (CR) and scale down each workflow to `0` before upgrading as shown in the following example:
```

bergerhoffer commented on modules/serverless-logic-upgrade-1-36-scaling-up-workflows-with-gitops-profile.adoc:
```diff
@@ -0,0 +1,35 @@
+
+.Procedure
+
+. Modify the `my-workflow.yaml` custom resources (CR) and scale up each workflow to `one` as shown in the following example:
```
```suggestion
. Modify the `my-workflow.yaml` custom resources (CR) and scale up each workflow to `1` as shown in the following example:
```

bergerhoffer reviewed (commented):
There are a handful of things that should be fixed before merging, if you don't mind!

kaldesai commented on serverless-logic/serverless-logic-managing-upgrades/serverless-logic-upgrading-operator-from-1-35-to-1-36.adoc:
```diff
@@ -0,0 +1,72 @@
+
+include::modules/serverless-logic-upgrade-1-36-scaling-up-workflows-with-gitops-profile.adoc[leveloffset=+2]
+
+include::modules/serverless-logic-verifying-the-1-36-0-upgrade.adoc[leveloffset=+2]
```
After rechecking the sections, I think that offset should be 1 because after completing all the steps, we verify the migration.

kaldesai commented:
@bergerhoffer, I have incorporated your suggestions, and the PR is now ready for the merge. Thank you for the feedback! :) If everything is okay, can you merge?

bergerhoffer commented on modules/serverless-logic-upgrade-1-36-deleting-workflows-with-dev-profile.adoc:
```diff
@@ -0,0 +1,20 @@
+
+.Procedure
+
+. Ensure you have a backup of all necessary Kubernetes resources, including `SonataFlow` custom resources (CRs), `ConfigMaps`, or any other related custom configurations.
```
```suggestion
. Ensure you have a backup of all necessary Kubernetes resources, including `SonataFlow` custom resources (CRs), `ConfigMap` resources, or any other related custom configurations.
```

wmedvede commented on modules/serverless-logic-upgrade-1-36-scaling-up-workflows-with-gitops-profile.adoc:
```diff
@@ -0,0 +1,35 @@
++
+[source,terminal]
+----
+$ oc patch sonataflow <workflow_name> -n <target_namespace> --type=merge -p '{"spec": {"podTemplate": {"replicas": 1}}}'
```
nitpick

kaldesai replied:
Fixed

bergerhoffer commented on modules/serverless-logic-verifying-the-1-36-0-upgrade.adoc:
```diff
@@ -0,0 +1,41 @@
+----
+$ oc get clusterserviceversion logic-operator-rhel8.v1.36.0 -n openshift-serverless-logic
+----
+.Expected output
```
Not critical, but this could be changed

```suggestion
.Example output
```

bergerhoffer reviewed (commented):
@kaldesai thanks for the updates! It looks great now. I added two small nits that I still found. I normally wouldn't hold up the merge for them, but I see that you're getting some additional feedback on the PR from an SME, so I'm going to hold off merging until you can take a look at that. You can fix mine if you have other updates to make, or it's also not a problem to leave those be since they're minor.

Feel free to ping me again once you've taken a look through those and are ready for merge. Thanks!

kaldesai commented on modules/serverless-logic-upgrade-1-36-finalizing-job-service.adoc:
```diff
@@ -0,0 +1,39 @@
+
+.Procedure
+
+. Identify the old `ReplicaSet` resource by running the following command:
```
@wmedvede We have mentioned "**....you can optionally delete the old `ReplicaSet`**" resource." at the top of the section. The entire procedure was optional, so I removed the `optional`

kaldesai commented on modules/serverless-logic-upgrade-1-36-scaling-down-workflows-with-gitops-profile.adoc:
```diff
@@ -0,0 +1,35 @@
++
+[source,terminal]
+----
+$ oc patch sonataflow <workflow_name> -n <target_namespace> --type=merge -p '{"spec": {"podTemplate": {"replicas": 0}}}'
```
Fixed

kaldesai commented:
@bergerhoffer 
Thanks for the review! I’ve addressed the minor items you raised and also incorporated the feedback from the SME. I double-checked everything, and it's all good to go now. Please feel free to proceed with the merge whenever it’s convenient for you. :)

wmedvede reviewed (approved):
@kaldesai Many thanks for the last nitpicks incorporation!!

bergerhoffer reviewed (commented):
Merging now!

bergerhoffer commented:
/cherrypick serverless-docs-1.35

bergerhoffer commented:
Once we CP #90790 to 1.36, this one should CP cleanly to 1.36.

bergerhoffer commented:
/cherrypick serverless-docs-1.36
